src/vanilla/CodeGeneratorPy.cs

```diff
@@ -56,8 +56,17 @@ public CodeGeneratorPy()
             var serviceClientTemplate = new ServiceClientTemplate { Model = codeModel };
             await Write(serviceClientTemplate, Path.Combine(folderName, codeModel.Name.ToPythonCase() + ".py"));
 
-            var versionTemplate = new VersionTemplate { Model = codeModel };
-            await Write(versionTemplate, Path.Combine(folderName, "version.py"));
+            // do we need to write out the version template file?
+            var versionPath = Path.Combine(folderName, "version.py");
+
+            // protect the version file
+            await Settings.Instance.Host.ProtectFiles(versionPath);
+
+            if( true != await Settings.Instance.Host.GetValue<bool?>("--keep-version-file")  ||  string.IsNullOrEmpty(await Settings.Instance.Host.ReadFile(versionPath)) ) {
```

Comment:
I went with the assumption that if the file wasn't there, you'd want to create it regardless of the intent of the consumer.